README.FT.ULFM.md

```diff
@@ -0,0 +1,424 @@
+
+- **mtl** Matching transport layer Used for MPI point-to-point messages on
+  some types of networks
+    - All "mtl" components are **disabled**
+
+- **coll** MPI collective algorithms
+    - "tuned", "basic", "nbc" modified to **handle errors**
+    - "cuda", "fca", "hcoll", "portals4" unmodified, **untested** (expect
+      unspecified post-failure behavior)
+
+- **osc** MPI one-sided communications
+    - Unmodified, **untested** (expect unspecified post-failure behavior)
+
+- **io** MPI I/O and dependent components
+    - _fs_ File system functions for MPI I/O
+    - _fbtl_ File byte transfer layer: abstraction for individual read/write
+      operations for OMPIO
+    - _fcoll_ Collective read and write operations for MPI I/O
+    - _sharedfp_ Shared file pointer operations for MPI I/O
+    - All components in these frameworks are unmodified, **untested**
+      (expect clean post-failure abort)
+
+- **vprotocol** Checkpoint/Restart components
+    - "pml-v", "crcp" unmodified, **untested**
+
+- **wait_sync** Multithreaded wait-synchronization object
+    - modified to **handle errors** (added a global interrupt to trigger all
+      wait_sync objects)
+___________________________________________________________________________
+
+Running ULFM Open MPI
+=====================
+
+## Building your application
+
+As ULFM is still an extension to the MPI standard, you will need to
+`#include <mpi-ext.h>` in C, or `use mpi_ext` in Fortran to access the
+supplementary error codes and functions.
+
+Compile your application as usual, using the provided `mpicc`, `mpif90`, or
+`mpicxx` wrappers.
+
+## Running your application
+
+You can launch your application with fault tolerance by simply using the
+provided `mpiexec`. Beware that your distribution may already provide a
+version of MPI, make sure to set your `PATH` and `LD_LIBRARY_PATH` properly.
+Note that fault tolerance is enabled by default in ULFM Open MPI; you can
+disable all fault tolerance systems by launching your application with
+`mpiexec --disable-recovery`.
+
+## Running under a batch scheduler
+
+ULFM can operate under a job/batch scheduler, and is tested routinely with
+both PBS and Slurm. One difficulty comes from the fact that many job
+schedulers will "cleanup" the application as soon as a process fails. In
+order to avoid this problem, it is preferred that you use `mpiexec`
+within an allocation (e.g. `salloc`, `sbatch`, `qsub`) rather than
+a direct launch (e.g. `srun`).
+
+## Run-time tuning knobs
+
+ULFM comes with a variety of knobs for controlling how it runs. The default
+parameters are sane and should result in very good performance in most
+cases. You can change the default settings with `--mca mpi_ft_foo <value>`.
+
+- `orte_enable_recovery <true|false> (default: true)` controls automatic
+  cleanup of apps with failed processes within mpirun. The default
+  differs from upstream Open MPI.
+- `mpi_ft_enable <true|false> (default: true)` permits turning off fault
```

Comment:
How about "prrte_abort_non_zero_exit" ?  If we don't use this flag, the job will be killed after failure.

Reply:
I get the appropriate behavior with --enable-recovery. You are however correct that the README needs updating with the new variable names for mca param files.